### PR TITLE
Add when parameter

### PR DESCRIPTION
**Pull request by drazisil**
Should address #13. Would love a code review.

Also de-duplicates the command step in favor of an optional parameter.

### CODE REVIEW

File: orb.yml [hunk 1]
```diff
@@ -50,3 +54,4 @@ commands:
                     -t << parameters.token >> \
                     -y << parameters.conf >> \
                     <<# parameters.flags >> -F << parameters.flags >> <</ parameters.flags >> \
+                when: << parameters.when >>
```
Comment (thomasrockhu) [inline orb.yml]: I think we will want this line duplicated on line 45 as well

Comment (drazisil) [reply orb.yml]: Missed that. Why do you have it duplicated like that? (not that versed in Orbs yet)

Comment (drazisil) [reply orb.yml]: Ah, now I see the diff.

**Comment by drazisil** [comment]
Should be better now, @thomasrockhu :)

**Comment by drazisil** [comment]
It's corrected with the line 37 conditional now, that line will only include the file if exists,  otherwise the command step will always run.

**Comment by mets11rap** [comment]
When can we expect this to be merged?

**Comment by mets11rap** [comment]
@thomasrockhu Could you merge this?

**Comment by thomasrockhu** [comment]
@drazisil unless I'm mistaken, this will still break functionality if no `file` parameter is supplied.

**Comment by drazisil** [comment]
@thomasrockhu I was told that `<<#parameters.file>> -f << parameters.file >>` would make it only include the `-f` flag if the file paramer exists.

I'm going on vacation for 2 weeks, and rather than delay this fix from landing in that time I have reverted the config to the prior behavior, with the `when:` modifier added.

We can revisit optimizing it later if you desire.

**Comment by mets11rap** [comment]
Not to sound naggy, but it's been a while @thomasrockhu

**Comment by thomasrockhu** [comment]
@mets11rap sorry for the wait! 

thanks again @drazisil 

I'll be merging PRs and releasing a new version today

**Comment by teohhanhui** [comment]
Bad indentation. Also fixed in #17

@thomasrockhu Could you take a look at #17? :smile:

**Pull request merged**